combat-trainer.lic

```diff
@@ -1402,7 +1400,7 @@ class SpellProcess
                      .select { |skill| Time.now > (@training_spells[skill]['cyclic'] ? @training_cyclic_timer : @training_cast_timer) }
                      .select { |skill| @training_spells[skill]['night'] ? UserVars.sun['night'] : true }
                      .select { |skill| @training_spells[skill]['day'] ? UserVars.sun['day'] : true }
-                     .sort_by { |skill| DRSkill.getxp(skill) }.first
+    needs_training = game_state.sort_by_rate_then_rank(needs_training).first
```

Comment:
Original logic is being updated to sort by rank, too. As well as to make that decision consistent throughout combat-trainer by using game_state.sort_by_rate_then_rank. If the decision algoritm should change in the future, there's now one spot to do it.